### PR TITLE
Use CPK to reduce the number of Tx necessary for market redeem

### LINKED ISSUE #382: Use CPK to reduce the number of Tx necessary for market redeem

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #382

### CODE REVIEW

File: app/src/services/cpk.ts [hunk 1]
```diff
@@ -325,6 +340,52 @@ class CPKService {
...
+    marketMaker,
+    conditionalTokens,
+  }: CPKRedeemParams): Promise<TransactionReceipt> => {
+    try {
+      const transactions = []
+      if (!isConditionResolved) {
+        transactions.push({
+          operation: CPK.CALL,
+          to: oracle.address,
+          value: 0,
```
Comment (fvictorio) [inline app/src/services/cpk.ts]: I'm starting to wonder if it wouldn't make sense to have `operation: CPK.CALL` and `value: 0` as default values. @cag how do you feel about that?

**Comment by fvictorio** [comment]
@mariano-aguero did you manually test this?

**Comment by mariano-aguero** [comment]
> @mariano-aguero did you manually test this?

Yes, I tested manually

**Comment by fvictorio** [comment]
> Yes, I tested manually

Doesn't work for me. I redeemed some tokens and didn't receive any collateral. How did you test it?

**Comment by mariano-aguero** [comment]
> > Yes, I tested manually
> 
> Doesn't work for me. I redeemed some tokens and didn't receive any collateral. How did you test it?

I added the transfer transaction

**Comment by fvictorio** [comment]
> I added the transfer transaction

@mariano-aguero Did you test the full workflow this time? Can I start reviewing it?

**Comment by mariano-aguero** [comment]
> > I added the transfer transaction
> 
> @mariano-aguero Did you test the full workflow this time? Can I start reviewing it?

Yes, sorry for that, I didn't know I had to transfer the collateral, you can try it now

File: app/src/services/cpk.ts [hunk 2]
```diff
@@ -417,6 +433,65 @@ class CPKService {
...
+          data: OracleService.encodeResolveCondition(questionId, numOutcomes),
+        })
+      }
+
+      const conditionId = await marketMaker.getConditionId()
+
+      transactions.push({
```
Comment (fvictorio) [inline app/src/services/cpk.ts]: Wouldn't this need to be inside `if (winningOutcome)`? I think it doesn't make sense to spend gas redeeming losing tokens.

**Comment by fvictorio** [comment]
Minor comment, but let's just merge this with or without it.

File: app/src/services/cpk.ts (hunk 2 above)
Comment (fvictorio) [reply app/src/services/cpk.ts]: Oh, nevermind, disregard that and just merge this.

**Pull request merged**